code/modules/mob/living/say.dm

```diff
@@ -129,10 +131,26 @@ var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 		to_chat(src, "<span class='warning'>You find yourself unable to speak!</span>")
 		return
 
-	if(message_mode != MODE_WHISPER) //whisper() calls treat_message(); double process results in "hisspering"
-		message = treat_message(message)
-		if(!message)
-			return
+	var/message_range = 6
+
+	if(message_mode == MODE_WHISPER)
+		message_range = 1
+		spans |= SPAN_ITALICS
+		log_whisper("[src.name]/[src.key] : [message]")
+		if(in_critical)
+			var/health_diff = round(-HEALTH_THRESHOLD_DEAD + health)
+			// If we cut our message short, abruptly end it with a-..
+			var/message_len = length(message)
+			message = copytext(message, 1, health_diff) + "[message_len > health_diff ? "-.." : "..."]"
+			message = Ellipsis(message, 10, 1)
+			message_mode = MODE_WHISPER_CRIT
+			succumb(1)

```

Comment:
You need to do this after sending the message, as otherwise the mob may not exist.